### PR TITLE
build/testing fixes for noetic compatibility

### PR DESCRIPTION
**Pull request by davefeilseifer**
updated python scripts and build dependencies to allow for building under noetic (and melodic) using python3.

passing prerelease test:

generate_prerelease_script.py   https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml   noetic default ubuntu focal amd64   --custom-repo     linux_networking__custom-2:git:https://github.com/PR2-prime/linux_networking.git:melodic-devel   --level 0   --output-dir ./

### CODE REVIEW

File: multi_interface_roam/package.xml
Comment (v4hn) [inline multi_interface_roam/package.xml]:
```suggestion
  <run_depend condition="$ROS_PYTHON_VERSION == 2">python-twisted-core</run_depend>
  <run_depend condition="$ROS_PYTHON_VERSION == 3">python3-twisted</run_depend> 
```

File: multi_interface_roam/package.xml
Comment (v4hn) [inline multi_interface_roam/package.xml]:
```suggestion
  <build_depend condition="$ROS_PYTHON_VERSION == 2">python-twisted-core</run_depend>
  <build_depend condition="$ROS_PYTHON_VERSION == 3">python3-twisted</run_depend> 
```

File: multi_interface_roam/src/multi_interface_roam/multi_interface_roam.py
Comment (v4hn) [inline multi_interface_roam/src/multi_interface_roam/multi_interface_roam.py]: I believe it's catkin's job to hardcode the version when files are installed with `catkin_install_python`.
Also you did not change the shebang line for any of the other scripts you edited, so please revert it here as well.

File: multi_interface_roam/src/multi_interface_roam/event.py
Comment (v4hn) [inline multi_interface_roam/src/multi_interface_roam/event.py]:
```suggestion
```

The import is not used in the file and it would break python2/3 compatibility.

**Review by v4hn — CHANGES_REQUESTED** [review]
Thank you for the patches and your work on pr2 noetic support! As a fellow PR2 maintainer I appreciate it. :+1: 

I don't think we have to break python2/3 compatibility here, so we don't need a new branch.
I added suggestions to keep your patch compatible with melodic.

File: multi_interface_roam/src/multi_interface_roam/multi_interface_roam.py
Comment (davefeilseifer) [reply multi_interface_roam/src/multi_interface_roam/multi_interface_roam.py]: sure

**Comment by v4hn** [comment]
Thanks!

**Comment by k-okada** [comment]
@v4hn @davefeilseifer Sorry for slow responce and thank you for contribution.
Are you sure that this PR works on noetic?

You put `  <run_depend condition="$ROS_PYTHON_VERSION == 3">python3-twisted</run_depend> ` that is nice, but I think you need `<package format="3"`.. May be we'd beter to run travis before we merge PR....https://github.com/PR2/linux_networking/pull/6
```
Traceback (most recent call last):
  File "/usr/bin/catkin", line 11, in <module>
    load_entry_point('catkin-tools==0.5.0', 'console_scripts', 'catkin')()
  File "/usr/lib/python3/dist-packages/catkin_tools/commands/catkin.py", line 272, in main
    catkin_main(sysargs)
  File "/usr/lib/python3/dist-packages/catkin_tools/commands/catkin.py", line 267, in catkin_main
    sys.exit(args.main(args) or 0)
  File "/usr/lib/python3/dist-packages/catkin_tools/verbs/catkin_build/cli.py", line 404, in main
    return build_isolated_workspace(
  File "/usr/lib/python3/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 296, in build_isolated_workspace
    workspace_packages = find_packages(context.source_space_abs, exclude_subspaces=True, warnings=[])
  File "/usr/lib/python3/dist-packages/catkin_pkg/packages.py", line 87, in find_packages
    packages = find_packages_allowing_duplicates(basepath, exclude_paths=exclude_paths, exclude_subspaces=exclude_subspaces, warnings=warnings)
  File "/usr/lib/python3/dist-packages/catkin_pkg/packages.py", line 147, in find_packages_allowing_duplicates
    parsed_package = parse_package_string(
  File "/usr/lib/python3/dist-packages/catkin_pkg/package.py", line 598, in parse_package_string
    raise InvalidPackage('The manifest contains invalid XML:\n%s' % ex, filename)
catkin_pkg.package.InvalidPackage: Error(s) in package '/home/user/ws_pr2/src/linux_networking/multi_interface_roam/package.xml':
The manifest contains invalid XML:

```

**Pull request merged**
